File: pdu/python/hako_srv_msgs/pdu_cdr_conv_NotifyAssetTimeResponse.py
from .pdu_pytype_NotifyAssetTimeResponse import NotifyAssetTimeResponse
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_NotifyAssetTimeResponse(writer: CdrWriter, src: NotifyAssetTimeResponse):
    writer.write_uint32(src.result_code)


def cdr_body_to_py_NotifyAssetTimeResponse(reader: CdrReader, dst: NotifyAssetTimeResponse):
    dst.result_code = reader.read_uint32()
    return dst


def py_to_cdr_NotifyAssetTimeResponse(src: NotifyAssetTimeResponse) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_NotifyAssetTimeResponse(writer, src)
    return writer.bytes()


def cdr_to_py_NotifyAssetTimeResponse(cdr_payload) -> NotifyAssetTimeResponse:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = NotifyAssetTimeResponse()
    return cdr_body_to_py_NotifyAssetTimeResponse(reader, dst)

File: pdu/javascript/hako_srv_msgs/pdu_cdr_conv_NotifyAssetTimeResponse.js
import { NotifyAssetTimeResponse } from './pdu_jstype_NotifyAssetTimeResponse.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduNotifyAssetTimeResponseConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { NotifyAssetTimeResponse } src
     */
    static to_cdr_body(writer, src) {
        writer.write_uint32(src.result_code);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { NotifyAssetTimeResponse } dst
     * @returns { NotifyAssetTimeResponse }
     */
    static cdr_body_to_js(reader, dst) {
        dst.result_code = reader.read_uint32();
        return dst;
    }

    /**
     * @param { NotifyAssetTimeResponse } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { NotifyAssetTimeResponse }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new NotifyAssetTimeResponse());
    }
}
